pkg/v3/plugin/ocr3.go

```diff
@@ -276,24 +218,20 @@ func (plugin *ocr3Plugin) ShouldTransmitAcceptedReport(_ context.Context, seqNr
 
 	plugin.Logger.Printf("%d upkeeps found in report for should transmit for sequence number %d", len(upkeeps), seqNr)
 
+	transmit := true
+
 	for _, upkeep := range upkeeps {
 		// if any upkeep in the report does not have confirmations from all coordinators, attempt again
-		allConfirmationsFalse := true
-		for _, coord := range plugin.Coordinators {
-			if coord.IsTransmissionConfirmed(upkeep) {
-				allConfirmationsFalse = false
-			}
-
-			plugin.Logger.Printf("checking transmit of upkeep '%s' %t", upkeep.UpkeepID, coord.IsTransmissionConfirmed(upkeep))
+		shouldTransmit := plugin.Coordinator.ShouldTransmit(upkeep)
+		if !shouldTransmit {
+			transmit = false
```

Comment:
The `allConfirmationsFalse` flag in the previous version of the code confused me here; is this new functionality correct? I.e. if even one upkeep receives a `false` response from `ShouldTransmit`, we should return false from this function?